format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -0,0 +1,192 @@
+        if (workSheet == null) {
+          failureCollector.addFailure("Sheet " + conf.getSheetValue() + " does not exist in the workbook.",
+                  "Specify a valid sheet.");
+          return null;
+        }
+
+        int sampleSizeInt = getSampleSizeInt();
+        // Row numbers are 0 based in POI
+        int rowStart = Math.min(0, workSheet.getFirstRowNum());
+        int rowEnd = Math.min(sampleSizeInt, workSheet.getLastRowNum());
+
+        int lastCellNumMax = 0;
+        List<String> columnNames = new ArrayList<>();
+        XlsInputFormatSchemaDetector schemaDetector = new XlsInputFormatSchemaDetector();
+        for (int rowIndex = rowStart; rowIndex <= rowEnd; rowIndex++) {
+          Row row = workSheet.getRow(rowIndex);
+          if (row == null) {
+            continue;
+          }
+          lastCellNumMax = Math.max(lastCellNumMax, row.getLastCellNum());
+
+          // Use the first row to get the column names
+          if (rowIndex == 0 && conf.getSkipHeader()) {
+            for (int cellIndex = 0; cellIndex < lastCellNumMax; cellIndex++) {
+              Cell cell = row.getCell(cellIndex, Row.MissingCellPolicy.RETURN_BLANK_AS_NULL);
+              columnNames.add(cell == null ? blankHeader : formatter.formatCellValue(cell, formulaEvaluator));
+            }
+            // Skip Header
+            continue;
+          }
+
+          for (int cellIndex = 0; cellIndex < lastCellNumMax; cellIndex++) {
+            Cell cell = row.getCell(cellIndex, Row.MissingCellPolicy.RETURN_BLANK_AS_NULL);
+            boolean isFirstRow = rowIndex == (conf.getSkipHeader() ? 1 : 0);
+            schemaDetector.reduceSchema(cellIndex, cell, isFirstRow);
+          }
+
+        }
+
+        // If some rows have more cells than the first row, add blank headers for the extra cells
+        if (lastCellNumMax > columnNames.size() && conf.getSkipHeader()) {
+          for (int i = columnNames.size(); i < lastCellNumMax; i++) {
+            columnNames.add(blankHeader);
+          }
+        }
+
+        // Set column names if header is not skipped
+        if (!conf.getSkipHeader()) {
+          for (int i = 0; i < lastCellNumMax; i++) {
+            columnNames.add(CellReference.convertNumToColString(i));
+          }
+        }
+
+        Schema schema = Schema.recordOf("xls", schemaDetector.getFields(
+                XlsInputFormatUtils.getSafeColumnNames(columnNames), conf.getOverride()));
+        return PathTrackingInputFormatProvider.addPathField(context.getFailureCollector(), schema, conf.getPathField());
+      }
+    }
+    return null;
+  }
+
+  private int getSampleSizeInt() {
+    // Note: Sample size is long, but we are casting it to int here. This is because the POI API uses int
+    // for row numbers, so we cannot support sample sizes larger than Integer.MAX_VALUE
+    long sampleSize = conf.getSampleSize() > Integer.MAX_VALUE ? Integer.MAX_VALUE : conf.getSampleSize();
```

Comment:
if there is POI limitation then `conf.getSampleSize()` should be types to `Integer` instead of `Long`.

Reply:
All source plugin have this as long, i have kept this as long to avoid any overflow issues while mapping of values.
I am clamping the value to Integer.MAX_VALUE, this will make sure we don't overflow.